repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java

```diff
@@ -0,0 +1,313 @@
+import org.alfresco.test_category.OwnJVMTestsCategory;
+import org.alfresco.util.BaseSpringTest;
+import org.alfresco.util.GUID;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Spy;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import javax.transaction.NotSupportedException;
+import javax.transaction.SystemException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+@Category(OwnJVMTestsCategory.class)
+@RunWith(SpringRunner.class) public class StorageClassTest extends BaseSpringTest
+{
+        private static final String DEFAULT_SC = "Default1";
+        private static final String DEFAULT_VALUE = "default";
+        private static final String TEST_NAMESPACE = "TestNameSpace";
+        private TransactionService transactionService;
+        private AuthenticationComponent authenticationComponent;
+
+        private NodeService nodeService;
+        private NodeRef rootNode;
+        @Spy ContentStore mockContentStore;
+        ContentService contentService;
+        ContentStore contentStore;
+
+        @Before
+        public void before()
+        {
+                nodeService = (NodeService) applicationContext.getBean("NodeService");
+                transactionService = (TransactionService) applicationContext.getBean("TransactionService");
+                this.authenticationComponent = (AuthenticationComponent) applicationContext.getBean("authenticationComponent");
+                this.contentService = (ContentService) this.applicationContext.getBean("contentService");
+                this.contentStore = (ContentStore) ReflectionTestUtils.getField(contentService, "store");
+
+                FileContentStore fileContentStore = new FileContentStore(applicationContext, this.contentStore.getRootLocation());
+                mockContentStore = spy(fileContentStore);
+
+                when(mockContentStore.getSupportedStorageClasses()).thenReturn(Set.of(DEFAULT_SC, "Azure", "S3"));
+                ReflectionTestUtils.setField(contentService, "store", mockContentStore);
+
+                this.authenticationComponent.setSystemUserAsCurrentUser();
+
+                StoreRef storeRef = new StoreRef(StoreRef.PROTOCOL_WORKSPACE, "testStoreRef");
+                if (!nodeService.exists(storeRef))
+                {
+                        storeRef = nodeService.createStore(storeRef.getProtocol(), storeRef.getIdentifier());
+                }
+                rootNode = nodeService.getRootNode(storeRef);
+        }
+
+        @After
+        public void afterTestMethod()
+        {
+                ReflectionTestUtils.setField(contentService, "store", contentStore);
```

Comment:
Again, why do we set an actual `contentStore`  `After` each method?

Reply:
We need to reset to the default content store after each test.